C4G.Unity/Assets/C4G/Core/Scripts/SheetsParsing/SheetsParsing.cs

```diff
@@ -8,97 +9,119 @@ public sealed class SheetsParsing
+            else if (parserBase == null)
+                error = "Sheets parsing error. Parser must be provided";
             else
             {
-                IList<object> headersRow = sheetData[0];
-
-                if (headersRow == null)
-                    error = "Sheets parsing error. Headers row must be not null";
-                else if (headersRow.Count != 2)
-                    error = "Sheets parsing error. Headers row length must be equal to two";
-                else if (!(headersRow[0] is string nameHeader) || nameHeader != NAME_HEADER)
-                    error = $"Sheets parsing error. First header must be equal to '{NAME_HEADER}'";
-                else
-                {
-                    if (!(headersRow[1] is string typeHeader) || typeHeader != TYPE_HEADER)
-                        error = $"Sheets parsing error. Second header must be equal to '{TYPE_HEADER}'";
-                    else
-                    {
-                        var firstDataRow = sheetData[1];
-
-                        if (firstDataRow == null)
-                            error = "Sheets parsing error. First data row must be not null";
-                        else if (firstDataRow.Count < 2)
-                            error = "Sheets parsing error. First data row length must be equal or greater than two";
-                        else
-                        {
-                            int firstDataRowLength = firstDataRow.Count;
-
-                            for (int rowIndex = 2; rowIndex < sheetData.Count; rowIndex++)
-                            {
-                                var dataRow = sheetData[rowIndex];
-
-                                if (dataRow == null)
-                                {
-                                    error = $"Sheets parsing error. '{rowIndex}' data row must be not null";
-                                    break;
-                                }
-
-                                int dataRowLength = dataRow.Count;
-
-                                if (dataRowLength != firstDataRowLength)
-                                {
-                                    error = $"Sheets parsing error. '{rowIndex}' data row length '{dataRowLength}' must be equal to first data row length '{firstDataRowLength}'";
-                                    break;
-                                }
-                            }
-                        }
-                    }
-                }
+                if (parserBase is HorizontalSheetParserBase)
+                    error = ValidateHorizontalFormat(sheetData);
+                else if (parserBase is VerticalSheetParserBase)
+                    error = ValidateVerticalFormat(sheetData);
             }
 
             return string.IsNullOrEmpty(error);
         }
+
+        private string ValidateHorizontalFormat(IList<IList<object>> sheetData)
```

Comment:
[Critical] I think we should move this validation on parser's side. Otherwise this class is going to expand dramatically with adding more and more persers to the project :)